libs/server-sdk/include/launchdarkly/server_side/bindings/c/sdk.h

```diff
@@ -0,0 +1,578 @@
+ * @param default_value The default value of the flag. Ownership is NOT
+ * transferred.
+ * @return The variation for the given context, or a copy of default_value if
+ * the flag is disabled in the LaunchDarkly control panel. The returned value
+ * must be freed using LDValue_Free.
+ */
+LD_EXPORT(LDValue)
+LDServerSDK_JsonVariation(LDServerSDK sdk,
+                          LDContext context,
+                          char const* flag_key,
+                          LDValue default_value);
+
+/**
+ * Returns the JSON value of a feature flag for a given flag key and context,
+ * and details that also describes the way the value was determined.
+ * @param sdk SDK. Must not be NULL.
+ * @param context The context. Ownership is NOT transferred. Must not be NULL.
+ * @param flag_key The unique key for the feature flag. Must not be NULL.
+ * @param default_value The default value of the flag. Ownership is NOT
+ * transferred.
+ * @param detail Out parameter to store the details. May pass LD_DISCARD_DETAILS
+ * or NULL to discard the details. The details object must be freed with
+ * LDEvalDetail_Free.
+ * @return The variation for the given context, or a copy of default_value if
+ * the flag is disabled in the LaunchDarkly control panel. The returned value
+ * must be freed using LDValue_Free.
+ */
+LD_EXPORT(LDValue)
+LDServerSDK_JsonVariationDetail(LDServerSDK sdk,
+                                LDContext context,
+                                char const* flag_key,
+                                LDValue default_value,
+                                LDEvalDetail* out_detail);
+
+/**
+ * Evaluates all flags for a context, returning a data structure containing
+ * the results and additional flag metadata.
+ *
+ * The method's behavior can be controlled by passing a combination of
+ * one or more options.
+ *
+ * A common use-case for AllFlagsState is to generate data suitable for
+ * bootstrapping the client-side JavaScript SDK.
+ *
+ * This method will not send analytics events back to LaunchDarkly.
+ *
+ * @param sdk SDK. Must not be NULL.
+ * @param context The context against which all flags will be evaluated.
+ * Ownership is NOT transferred. Must not be NULL.
+ * @param options A combination of one or more options. Pass
+ * LD_ALLFLAGSSTATE_DEFAULT for default behavior.
+ * @return An AllFlagsState data structure. Must be freed with
+ * LDAllFlagsState_Free.
+ */
+LD_EXPORT(LDAllFlagsState)
+LDServerSDK_AllFlagsState(LDServerSDK sdk,
+                          LDContext context,
+                          enum LDAllFlagsState_Options options);
+
+/**
+ * Frees the SDK's resources, shutting down any connections. May block.
+ * @param sdk SDK.
+ */
+LD_EXPORT(void) LDServerSDK_Free(LDServerSDK sdk);
+
+typedef struct _LDServerDataSourceStatus* LDServerDataSourceStatus;
+
+/**
+ * Enumeration of possible data source states.
+ */
+enum LDServerDataSourceStatus_State {
```

Comment:
I admit these enums are quite ugly - but needed to rename so they don't clash with the client and can be independently changed.

Welcome to suggestions..